File: LeetCodeProblemsCMake/Problems/LeetCode3713LongestBalancedSubstringI.cpp
//
// Created by saeed on 2/12/26.
//

#include "LeetCode3713LongestBalancedSubstringI.h"

#include <vector>
using namespace std;

int LeetCode3713LongestBalancedSubstringI::longestBalanced(string s) {
    int length = static_cast<int>(s.length());
    int maxLength{1};
    int target{0};
    bool balanced{false};
    for (int i = 0; i < length; ++i) {
        auto occurrences = vector<int>(26);
        for (int j = i; j < length; ++j) {
            target = ++occurrences[s[j] - 'a'];
            balanced = true;
            for (auto num: occurrences) {
                if (num == 0) {
                    continue;
                }
                if (num != target) {
                    balanced = false;
                    break;
                }
            }
            if (balanced) {
                maxLength = max(maxLength, j - i + 1);
            }
        }
    }
    return maxLength;
}

File: LeetCodeProblemsGo/LeetCode3713LongestBalancedSubstringI/LeetCode3713LongestBalancedSubstringI.go
package LeetCode3713LongestBalancedSubstringI

import "fmt"

func longestBalanced(s string) int {

	maxLength := 1
	length := len(s)
	target := 0
	balanced := false
	for i := 0; i < length; i++ {
		occurances := make([]int, 26)
		for j := i; j < length; j++ {
			occurances[s[j]-'a']++
			target = occurances[s[j]-'a']
			balanced = true
			for _, num := range occurances {
				if num == 0 {
					continue
				}
				if num != target {
					balanced = false
					break
				}
			}
			if balanced {
				maxLength = max(maxLength, j-i+1)
			}
		}
	}
	return maxLength
}
func Calculate() {
	s := "zzabccy"
	fmt.Println(longestBalanced(s))
}
